src/shared/mount-util.c

```diff
@@ -77,13 +77,25 @@ int umount_recursive(const char *prefix, int flags) {
         return n;
 }
 
-static int get_mount_flags(const char *path, unsigned long *flags) {
-        struct statvfs buf;
+/* Get the mount flags for the mountpoint at "path" from "table" */
+static int get_mount_flags(const char *path, unsigned long *flags, struct libmnt_table *table) {
+        struct libmnt_fs *fs = NULL;
+        const char *opts = NULL;
+        int r = 0;
+
+        fs = mnt_table_find_target(table, path, MNT_ITER_FORWARD);
+        if (!fs) {
+            r = -ENOENT;
```

Comment:
why not fall back to statvfs in this case?

alternatively: this deserves an explicit log message, explaining what happens, subsuming that in to the warning you have below is just weird, because one might thing some file or directory might be missing